src/js/tasks-function.js

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable max-classes-per-file */
+
+class Tasks {
+  constructor(description, completed = false, index) {
+    (this.description = description);
+    (this.completed = completed);
+    (this.index = index);
+  }
+}
+
+class TaskStatus {
+  constructor() {
+    this.completed = false;
+  }
+
+  checked = (task) => {
+    task.completed = true;
+  };
+
+  unchecked = (task) => {
+    task.completed = false;
+  };
+}
+
+class Todolist {
+  constructor() {
+    this.taskData = [];
+  }
+
+  displayTask = () => {
+    const taskList = document.querySelector('.task-list');
+    if (taskList) {
+      taskList.innerHTML = '';
+    }
+    this.taskData.forEach((elem) => {
+      const li = [];
+      li[elem.index] = document.createElement('li');
+      li[elem.index].setAttribute('id', elem.index);
+      li[elem.index].classList.add('flex');
+      const label = [];
+      label[elem.index] = document.createElement('label');
+      label[elem.index].textContent = elem.description;
+      label[elem.index].classList.add('label');
+      label[elem.index].setAttribute('id', elem.index);
+      label[elem.index].contentEditable = true;
+      const inputBox = [];
+      inputBox[elem.index] = document.createElement('input');
+      inputBox[elem.index].setAttribute('type', 'checkbox');
+      inputBox[elem.index].setAttribute('id', elem.index);
+      inputBox[elem.index].classList.add('checkbox');
+      const moveBtn = [];
+      moveBtn[elem.index] = document.createElement('button');
+      moveBtn[elem.index].setAttribute('id', elem.index);
+      moveBtn[elem.index].innerHTML = '<i class="fa-solid fa-ellipsis-vertical ellipsis"></i>';
+
+      li[elem.index].append(
+        inputBox[elem.index],
+        label[elem.index],
+        moveBtn[elem.index],
+      );
+      taskList.append(li[elem.index]);
+      li[elem.index].addEventListener('click', () => {
+        li[elem.index].style.backgroundColor = '#ff98008f';
+      });
+      label[elem.index].addEventListener('click', (e) => {
+        e.target.nextSibling.innerHTML = '<i class="fa-solid fa-trash-can"></i>';
+        e.target.nextSibling.style.cursor = 'pointer';
+
+        e.target.nextSibling.addEventListener('click', () => {
+          li[elem.index].remove();
+          this.removetask(elem.index);
+        });
```

Comment:
- [ ] Due to the error in indexing, I noticed that whenever I attempt to remove an item, I could remove two items with the same index key. I would advise you to fix the issue with the indexing and ensure only the desired task is removed from the todo list. 💻 👍